### PR TITLE
feature/B2C-303-ApiVersioningImprovements

### PR DESCRIPTION
**Pull request by smneal**
- Added settings to support API versioning of Customers and Digital Identities API
- Updated Notify package to use 4.0.1 available from NuGet.org as previous source was shutdown
- Fixed missing tags which are now mandatory in deployments

### CODE REVIEW

**Review by BAAdeosun — APPROVED** [review]

**Review by soharadesign — APPROVED** [review]
Looks good!

It would be best to add a config value for NcsDssContactDetailsApiVersion too, as that could also be versioned independently in the future.

"NcsDssContactDetailsApiVersion": "v1"

Needs to be used when the NcsDssCreateContactApiUrl URL is called.

File: src/B2CAzureFunc/local.settings.json [hunk 1]
```diff
@@ -30,11 +30,13 @@
     "ncsdsssearchapiurl": "https://at.api.nationalcareersservice.org.uk/search/api/",
     "ncsdssapikey": "92f1fe387ea14bab9567403c4c1d5e72",
     "OcpApimSubscriptionKey": "92f1fe387ea14bab9567403c4c1d5e72",
-    "ncsdsssearchapiversion": "v2",
+    "NcsDssSearchApiVersion": "v3",
```
[the remark this answers is not in the document]
Comment (smneal) [reply src/B2CAzureFunc/local.settings.json]: Sorted while adding NcsDssContactDetailsApiVersion

File: tests/B2CAzureFunc.Tests/local.settings.json [hunk 2]
```diff
@@ -29,11 +29,13 @@
     "ncsdsssearchapiurl": "https://at.api.nationalcareersservice.org.uk/search/api/",
     "ncsdssapikey": "92f1fe387ea14bab9567403c4c1d5e72",
     "OcpApimSubscriptionKey": "92f1fe387ea14bab9567403c4c1d5e72",
-    "ncsdsssearchapiversion": "v2",
+    "NcsDssSearchApiVersion": "v3",
```
[the remark this answers is not in the document]
Comment (smneal) [reply tests/B2CAzureFunc.Tests/local.settings.json]: Sorted while adding NcsDssContactDetailsApiVersion

**Comment by smneal** [comment]
Thanks @soharadesign. I've added a setting for NcsDssContactDetailsApiVersion and sorted the NcsDssSearchApiVersion out too. Cheers

**Review by soharadesign — APPROVED** [review]

**Pull request merged**